fairlearn/metrics/_roc_auc.py

```diff
@@ -0,0 +1,328 @@
+# Copyright (c) Microsoft Corporation and Fairlearn contributors.
+# Licensed under the MIT License.
+
+import matplotlib.pyplot as plt
+
+from sklearn.metrics import roc_curve, auc, roc_auc_score
+from sklearn.metrics import RocCurveDisplay
+
+from fairlearn.metrics import MetricFrame
+
+class RocAuc:
+    """
+    Provides utilties for generating AUC scores, ROC curves
+    and plotting ROC curves grouped by sensitive feature. Depends on
+    the Fairlearn MetricFrame class to split the input by sensitive feature.
```

Comment:
@romanlutz Will that markdown automatically create a link to the API in the docs when I build the docs?

Reply:
Yes! It's restructured text (ReST) rather than markdown, but that's exactly what it does. I go back to the ReST guide quite a lot these days: https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html